ddtrace/appsec/iast/taint_sinks/_base.py

```diff
@@ -74,45 +84,163 @@ def report(cls, evidence_value="", sources=None):
-                            )
-                        )
+            if not frame_info:
+                return None
+
+            file_name, line_number = frame_info
+
+            # Remove CWD prefix
+            if file_name.startswith(CWD):
+                file_name = os.path.relpath(file_name, start=CWD)
+
+            if _is_evidence_value_parts(evidence_value):
+                evidence = Evidence(valueParts=evidence_value)
+            elif isinstance(evidence_value, (str, bytes, bytearray)):
+                evidence = Evidence(value=evidence_value)
+            else:
+                log.debug("Unexpected evidence_value type: %s", type(evidence_value))
+                evidence = ""
+
+            if not cls.is_not_reported(file_name, line_number):
+                # not not reported = reported
+                return None
+
+            _set_metric_iast_executed_sink(cls.vulnerability_type)
 
-                    else:
-                        report = IastSpanReporter(
-                            vulnerabilities={
-                                Vulnerability(
-                                    type=cls.vulnerability_type,
-                                    evidence=evidence,
-                                    location=Location(path=file_name, line=line_number, spanId=span.span_id),
-                                )
-                            }
+            report = _context.get_item(IAST.CONTEXT_KEY, span=span)
+            if report:
+                report.vulnerabilities.add(
+                    Vulnerability(
+                        type=cls.vulnerability_type,
+                        evidence=evidence,
+                        location=Location(path=file_name, line=line_number, spanId=span.span_id),
+                    )
+                )
+
+            else:
+                report = IastSpanReporter(
+                    vulnerabilities={
+                        Vulnerability(
+                            type=cls.vulnerability_type,
+                            evidence=evidence,
+                            location=Location(path=file_name, line=line_number, spanId=span.span_id),
                         )
-                    if sources:
-                        report.sources = {Source(origin=x.origin, name=x.name, value=x.value) for x in sources}
-                    _context.set_item(IAST.CONTEXT_KEY, report, span=span)
+                    }
+                )
+            if sources:
+                report.sources = {Source(origin=x.origin, name=x.name, value=x.value) for x in sources}
+
+            redacted_report = cls._redacted_report_cache.get(hash(report), lambda: cls._redact_report(report), False)
+            _context.set_item(IAST.CONTEXT_KEY, redacted_report, span=span)
+
+    @classmethod
+    def _extract_sensitive_tokens(cls, report):
+        # type: (IastSpanReporter) -> Set[str]
+        log.debug("Base class VulnerabilityBase._extract_sensitive_tokens called")
```

Comment:
log? isn't it better a NotImplementedError exception?

Reply:
Done but it will be capture anyway upwards.